examples/shell/cmd_device.cpp

```diff
@@ -0,0 +1,374 @@
+    return error;
+}
+
+static CHIP_ERROR ConfigGetVendorId(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint16_t value16;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "VendorId:        ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetVendorId(value16));
+    streamer_printf(sout, "%04x", value16);
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetProductId(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint16_t value16;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "ProductId:       ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetProductId(value16));
+    streamer_printf(sout, "%04x", value16);
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetProductRevision(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint16_t value16;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "ProductRevision: ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetProductRevision(value16));
+    streamer_printf(sout, "%04x", value16);
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetSerialNumber(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    char buf[ConfigurationManager::kMaxSerialNumberLength];
+    size_t bufSize;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "SerialNumber:    ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetSerialNumber(buf, sizeof(buf), bufSize));
+    streamer_print_hex(sout, reinterpret_cast<const uint8_t *>(buf), bufSize);
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetDeviceId(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint64_t value64;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "DeviceId:        ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetDeviceId(value64));
+    streamer_print_hex(sout, reinterpret_cast<const uint8_t *>(&value64), sizeof(value64));
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetDeviceCert(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint8_t buf[512];
+    size_t bufSize;
+
+    if (printHeader)
+    {
+        streamer_printf(sout, "DeviceCert:      ");
+    }
+    SuccessOrExit(error = ConfigurationMgr().GetDeviceCertificate(buf, sizeof(buf), bufSize));
+    streamer_print_hex(sout, const_cast<const uint8_t *>(buf), bufSize);
+
+exit:
+    return ConfigGetDone(sout, error);
+}
+
+static CHIP_ERROR ConfigGetDeviceCaCerts(bool printHeader)
+{
+    CHIP_ERROR error  = CHIP_NO_ERROR;
+    streamer_t * sout = streamer_get();
+    uint8_t buf[512];
```

Comment:
these stack buffers are somewhat unfortunate.  streamer or THE_SHELL can't help with this?

Reply:
They are a consequence of the `DeviceLayer::ConfigurationManager` APIs as far as I can tell.  We can improve in a separate PR.  It would be bit less severe if all the fields had kMaxSize defined in `ConfigurationManager`.

Reply:
yeah, I see that.  if these functions aren't re-entrant, how about move the buffer (make it big enough for all these cases) to the shell or to a static variable?

Reply:
Not sure I agree.  Stack is purely temporary, whereas static storage permanently takes away from all modules on the system.  Moving these to a shared static increases the cost of integrating these shell commands into your firmware.

Reply:
good point

Reply:
Ended up fixing these "the right way" according to #1586